packaging/rpm/zrepl.spec

```diff
@@ -0,0 +1,100 @@
+# This Spec file packages pre-built artifacts in the artifacts directory
+# into an RPM. This means that rpmbuild won't actually build any new artifacts
+#
+# Read the Makefile first, then come back here.
+
+# Add workaround for "No build ID note found" error during %%install
+%undefine _missing_build_ids_terminate_build
+
+
+# Disable debuginfo for RedHat/CentOS builds to avoid errors due to empty
+# debuginfo files, see also: https://fedoraproject.org/wiki/Packaging:Debuginfo
+%if 0%{?rhel}
+%global debug_package %{nil}
+%endif
+
+
+# Global meta data
+Version:          SUBSTITUTED_BY_MAKEFILE
+%global common_description %{expand:
+zrepl is a one-stop, integrated solution for ZFS replication.}
+
+
+Name:             zrepl
+Release:          1%{?dist}
+Summary:          One-stop, integrated solution for ZFS replication
+License:          MIT
+URL:              https://zrepl.github.io/
+# Source: we use rpmbuild --build-in-tree => no source
+BuildRequires:    git >= 2
+BuildRequires:    systemd
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
+
+%description
+%{common_description}
+
+$prep
+# we use rpmbuild --build-in-tree => no prep or setup
+
+%build
+# we don't actually need to build anything here, that has already been done by the Makefile
+
+# Correct the path in the systemd unit file
+sed s:/usr/local/bin/:%{_bindir}/:g dist/systemd/zrepl.service > artifacts/rpmbuild/zrepl.service
+
+# Generate the default configuration file
+cat > artifacts/rpmbuild/zrepl.yml <<-EOF
+	global:
+	  logging:
+	    # use syslog instead of stdout because it makes journald happy
+	    - type: syslog
+	      format: human
+	      level: warn
+
+	jobs:
+	#  - name: foo
+	#    type: bar
+
+	# see %{_datadir}/doc/zrepl/examples
+	# or https://zrepl.github.io/configuration/overview.html
+EOF
```

Comment:
Given that the spec file is not standalone anymore anyway, the default config file should be added to the git repository and copied (with replacing the path to the examples using sed) analog to the systemd file. This would ensure that the same config file is distributed in all packages (rpm, deb, ...).